system/jlib/jstats.cpp

```diff
@@ -1988,9 +2000,58 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
             cur.visit(visitor);
     }
 
+    virtual bool refreshAggregates(std::vector<StatisticKind> & aggregateKinds, std::vector<unsigned __int64> & totals) override
+    {
+        assertex(aggregateKinds.size()==totals.size());
+
+        bool updated = false;
+        if (queryScopeType()==SSTsubgraph)
+        {
+            ForEachItemIn(i, stats)
+            {
+                Statistic & stat = stats.element(i);
+                StatisticKind kind = stat.queryKind();
+                auto iteratorVec = std::find(aggregateKinds.begin(), aggregateKinds.end(), kind);
+                if (iteratorVec!=aggregateKinds.end())
+                {
+                    unsigned pos = iteratorVec-aggregateKinds.begin();
+                    StatsMergeAction mergeAction = queryMergeMode(kind);
+                    totals[pos] = mergeStatisticValue(totals[pos], stat.queryValue(), mergeAction);
+                    updated = true;
+                }
+            }
+        }
+        else
+        {
+            std::vector<unsigned __int64> childTotals(aggregateKinds.size());
+            for (auto & child : children)
+            {
+                if (child.refreshAggregates(aggregateKinds, childTotals))
+                    updated = true;
+            }
+            if (updated)
+            {
+                std::vector<unsigned __int64>::iterator totalIter = totals.begin();
+                std::vector<unsigned __int64>::iterator subTotalIter = childTotals.begin();
+                std::vector<StatisticKind>::iterator kindIter = aggregateKinds.begin();
+                while (totalIter != totals.end())
+                {
+                    StatsMergeAction mergeAction = queryMergeMode(*kindIter);
+                    updateStatistic(*kindIter, *subTotalIter, mergeAction);
```

Comment:
I think this should probably replace rather than merge - since this is the new aggregated value.

Reply:
I was thinking StatsMergeAction would help to keep track of peaks, such as PeakSpillSize. What do you think?  Is simple aggregation enough (replace action) or should we try to handle other types such as peaks?